baybe/parameters/substance.py

```diff
@@ -156,19 +153,30 @@ def values(self) -> tuple:
         # for Python 3.7 or higher
         return tuple(self.data.keys())
 
+    # TODO: @Roya not sure if dynamic import in cached_property is a good practice
```

Comment:
maybe, but is there a reason why you've set the error to `warn` below? Seems if this would be `raise` then this cached_property could never accidentally be created and things should be fine imo

In fact im wondering whether there is any place where is should not be `raise` and why? (see in some other places you also use `warn`)

Reply:
My feeling is that the `warn` functionality might not be suitable for the `chem` dependencies but maybe for other - `simulation` maybe? But I agree, `raise` feels more reasonable for me here as well.

Reply:
Makes sense 😊